---
toc: true
toc_label: "Lab Handbook"
toc_icon: "bacteria"
toc_sticky: true
permalink: /handbook/
title: ""
---

# Lab values and expectations

## Scientific goals

Our goal is to use population genomics and statistical methods to study the epidemiology and evolution of bacterial pathogens of human and animal populations, including addressing the pressing public health threat of antimicrobial resistance. The lab employs computational tools using microbial whole genome sequencing data to reconstruct pathogen transmission, investigate adaptation during infection and transmission, and inform the development of novel diagnostics and interventions. 

We also aim to train scientists who work both independently and collaboratively to study the impact of pathogen ecology and evolution on public health using microbial genomic data. Trainees are encouraged to devise projects aligned with their career goals and establish collaborations both inside and outside the lab.

## Diversity, equity, and inclusion

In the Mortimer Lab, we are committed to fostering a diverse scientific workforce through our research, mentorship, and teaching activities within the lab, across the university, and in the community. Lab members agree to the following actions: 

- **Regularly reflect on the impact of our research on our community.** The burden of infectious diseases disproportionately impacts minoritized communities in the United States and under resourced settings globally. In our [lab meetings](#lab-meeting), we discuss scientific literature related to diversity, equity, and inclusion (DEI). [Reproducible research](#reproducible-research) and equitable access to results is also a lab value, and we commit to posting preprints, publishing in open access journals, and sharing data and code for published manuscripts. 

- **Increase exposure to and representation in microbiology and computational biology.** Graduate student and postdoctoral trainees are expected to participate in mentorship of other lab members. As a lab, we will engage in at least one volunteer or outreach activity in the Athens-Clarke County area yearly. Lab members are also encouraged to participate in community outreach activities individually. Some activities we recommend include [Letters to a Pre-Scientist](https://prescientist.org/) and [Skype a Scientist](https://www.skypeascientist.com/).

- **Recruit and retain lab members from underrepresented backgrounds.** The unspoken rules of academia are often barriers to first generation and minoritized students. This lab handbook includes information to help trainees in the lab, prospective lab members, and the UGA research community. To ensure that the opportunity to work in our lab is equitable, we do not have volunteer positions. If you are an undergraduate at University of Georgia, our lab policy is that undergraduate students who work in the lab will either receive undergraduate course credit or be paid hourly. We also participate in summer internship programs for undergraduates like [NSURP](https://nsurp.org/). 

## Reproducible research

Our goal is to make all research from the lab reproducible, meaning given the raw data and our code someone else could rerun the analyses from start to finish and get the same results. Additionally, we need to keep a record of when and why code is written or changed.

Making your workflow reproducible from the start provides several benefits. Well documented data, pipelines, and scripts make writing the methods section of your manuscript and complying with academic journal and funding agency requirements for data sharing straightforward. Additionally, the scripts and pipelines written for one project may be easily adapted for your next project. Lastly, when you inevitably have to rerun most of your analyses because of an update in the data or a change in one piece of the code, you will be thankful that your past self made this simple!

When we have agreement from any outside collaborators, all manuscripts in the lab will be posted first as a preprint and then published in an open access journal after peer review. If you do not have access to a published article from the lab, please email to get a PDF.

## Wellness

In addition to the stress that comes with academic training, your wellness and productivity may be impacted by health, home, and family matters. We are committed to supporting all trainees throughout their time in the lab, including encouraging healthy work life balance, mental health, and pointing you to [wellness resources](#wellness-resources) available here at UGA.

## General expectations

1. Treat other lab members and collaborators with respect
2. Let me know if you are not being treated with respect or something about the lab environment is not working for you
3. Conduct research responsibly, reproducibly, and rigorously
4. Thoroughly [document](#data-storage-and-sharing) and share your work
5. Ask for feedback from me, collaborators, and other lab members in a timely manner
6. Contribute to the lab environment: share your expertise with others, participate in lab meetings, and offer feedback on experimental design, code, figures, and drafts

# Lab day to day

## Work hours and time off

### Hours and remote work

One of the perks of working in a primarily computational lab is the ability to work when and where is best for you, and I won't be keeping tabs on exactly when you are in the lab as long as you are making progress on your work. However, there are many benefits to socializing with and learning from other members of the lab and department, so I would encourage everyone to physically be in the lab at least three days per week during normal working hours (overlapping 11am-3pm). To that end, if there are ways I can make the lab a more comfortable space for you to work, please let me know. I expect that hourly employees will accurately report their time.

The above does not apply to undergraduate students who need to work around their class schedules, may not work more than 20 hours per week, and should only be in lab during normal work hours. Students doing research for credit should plan to work at least 45 hours per credit hour in a given semester. Undergrads should discuss their planned schedule and work hours with me at the beginning of the semester.

### Vacation

I will not be in the lab on official [UGA paid holidays](https://www.uga.edu/holiday-schedule), and I hope that everyone will take this time off as well. If you observe holidays that are not on UGA's holiday calendar and will need time off, please let me know.

UGA benefits eligible employees and postdoctoral researchers accrue vacation hours at a rate of 10 hours per month. While graduate students do not officially accrue any leave, I also encourage graduate students to take time off, and you should feel free to take 15 days off in addition to UGA holidays.  I do not need to approve the dates of your vacation in advance, but I would appreciate it if you let me and the rest of the lab know when you plan to be on vacation. When you are on vacation, I do not expect you to respond to lab-related communications. Work or career-related travel (like attending conferences) is work, and it does not count as vacation.

[UGA policies on annual leave for benefits-eligible employees](https://hr.uga.edu/Current_Employees/Leave/annual_leave/)

[Postdoctoral Researcher specific policies](https://policies.uga.edu/Human-Resources/Time-away-from-work/Annual-Leave/Postdoctoral-Researchers/)

### Sick time

Please do not come to lab if you are sick and take the time you need to rest and recover! If you are feeling well enough to work but may still be contagious, please work remotely.

[UGA policy on sick leave for benefits-eligible employees](https://hr.uga.edu/Current_Employees/Leave/sick_leave/)

## Lab communication

The lab primarily communicates via a [slack workspace](https://mortimer-lab.slack.com). If you have a question for me or another lab member, please first consider if it can be posted in a public channel. It's likely that someone else has the same question now (or in the future) and that multiple lab members will be able to help you with your problem. Posting in a channel, rather than a direct message, will allow the discussion to be searchable in the future.

If my office door is open, please feel free to come in. If I'm out of office or my office door is closed, it is best to send me a message on slack. I expect that everyone checks in on the lab slack a few times a day on days you are working. You do not need to respond to messages at night, on the weekend, or on vacation even if other lab members are working during those times.

The lab also has a shared calendar (which doesn't exist just yet). Everyone should feel free to add or update events on the calendar that may be of interest to the lab, including lab meetings, seminars, and conference abstract deadlines. I will also keep the calendar up to date with days that I will be traveling for work or on vacation. 

If you would like to share your planned vacations on the lab calendar or update your slack status when you are away or sick, you are welcome to do so. However, I want to emphasize that this is not a requirement, and while you should let me know when you will be away from the lab, you do not have to share your specific schedule or reasons for being out with the whole lab.

## Meetings

### Individual meetings

For junior trainees, we will have weekly one-on-one meetings; after one year in the lab, we can reduce the frequency of meetings to every other week (or keep meeting weekly!). Prior to our meetings please make an agenda for the meeting and share it with me on slack or in your lab notebook. This time is yours to use as you like, but I suggest including an update on what you accomplished in the last week and a plan for the next week's work. It can also be helpful to share slides with data you'd like to discuss. 

### Yearly individual goals meeting

In the spring, we will use one of our individual meetings to complete any required yearly evaluations and discuss your long-term research and career goals. If you do not have a required written evaluation as part of your training program, I would suggest using an [individual development plan](https://myidp.sciencecareers.org/).

### Lab meeting

We have a weekly, one hour lab meeting. We will use this the first 10-15 minutes for general lab updates, and we will use the remaining time for research presentations, practice talks or poster review, discussion of journal articles, tutorials on computational or other scientific skills, or discussion of broader topics which put our work in context with our community, including outreach, advocacy, and DEI. At the beginning of the fall, spring, and summer, we will reschedule our weekly lab meetings to best accommodate everyone's schedules.

### Yearly lab goals meeting

In the fall, we will schedule a half day lab meeting to discuss our projects and goals as a lab. During this time, I will also give a 'state of the lab' update on lab achievements, funding, and lab directions for the next year. We will also use this time to review this handbook to make sure it is up-to-date, useful, and reflects our shared values and expectations.

## Seminars

### Department of Microbiology Seminars

[Schedule](https://mib.uga.edu/events/all)

# Resources

## Lab space

The lab is located in the Poultry Diagnostic and Research Center Building (953 College Station Rd) in Room 121, which can be accessed from the main lobby. 

## Computers

The lab will provide a computer, monitor, external hard drive, and workspace for anyone working full-time in the lab (staff, graduate students, postdocs). If you are an undergraduate or rotation student in the lab who does not have a usable laptop for your work, please come talk to me; we will find a way to get you a computer with either lab or university resources.

## Slack

The lab's [slack workspace](https://mortimer-lab.slack.com) is the main method of lab communication. You will be automatically added to several channels when you join, but please also check out the channel browser and join any other public channels related to your work.

## Calendar

Coming soon!

## GitHub

We share all project related code through the lab [GitHub organization](https://github.com/mortimer-lab).

## OneDrive

UGA provides up to 5TB of storage through [Microsoft OneDrive](https://ugamail.uga.edu/onedrive_business/). We have a shared lab folder, and I would also encourage you to use your individual account for automatic back up of your work-related files. 

## GACRC Sapelo2 Cluster

The majority of your work will take place on UGA's research cluster Sapelo2. Once your training progresses to work on the cluster, please request an [account](https://gacrc.uga.edu/accounts/). Depending on your prior experience working on clusters, you may be required to attend one or more trainings before being given access.

Please regularly review [best practices](https://wiki.gacrc.uga.edu/wiki/Best_Practices_on_Sapelo2) for working on Sapelo2. 

## Academic journal access

For some academic journals, you may need to access articles through UGA's subscriptions. I find that the easiest way to get access to an article through UGA is to go to the [library homepage](https://www.libs.uga.edu/) and use the Google Scholar search option from there.

## Wellness resources

- [UGA Student Care and Outreach](https://sco.uga.edu/resources-health/)

# Joining and leaving the lab

## Lab Onboarding

1. Complete new hire checklists provided by PDRC
2. Complete required training available through PEP
3. Make sure you have access to the lab resources above. If I've forgotten to give you access to one of them, please remind me!
4. Send me a short bio and a picture of yourself for the lab website

Optional onboarding for beginners in bioinformatics:
5. Complete tutorials in [unix](#unix), [git](#git), and [python](#python)
6. Try to complete as many problems as you can through [Rosalind](https://rosalind.info/problems/locations/). During our first few weekly meetings, we can discuss where you got stuck, what you think the problem is, and work through a solution together. Feel free to discuss the problems with other lab members as well!

## Before you go

1. Make sure that all published and unpublished data is available according to the data storage and sharing section below
2. Double check that you have all the data you need from lab-owned computers and contact IT to have them reset for a future lab member's use
3. Clean your workspace
4. Schedule a meeting with me and anyone who will be working on your projects going forward to make sure we know where everything stands
5. Share your contact information, so we can stay in touch!

# Data storage and sharing

## Lab Notebooks

### Electronic lab notebooks

In this lab, we use [Obsidian](https://obsidian.md/) as an electronic lab notebook. Please create an Obsidian vault in your OneDrive account and share the folder with me. To save your Obsidian vault in OneDrive, you will need the desktop app available [here](https://ugamail.uga.edu/onedrive_business/). Also, check out the numerous plugins available for Obsidian.

For each project, please create a folder in your notebook which contains the following: an up-to-date note listing locations for raw data, analyses, results, presentations, and manuscripts related to your project; a note outlining research related tasks that are in progress, still to do, or done; and any other project related notes. Ideally, all analyses should be 100% reproducible, so code, commands, and data should primarily reside on the cluster and changes should be tracked using version control with git and pushed to the lab's GitHub regularly. If you receive data from a collaborator, please take detailed notes about who sent you the data, when you received the data, and what they had to say about the data.

Additionally, your Obsidian vault can be a useful place to keep notes on literature, our individual meetings, ideas for future projects and fellowship applications, and things like tricky software installations. If you have notes that you think would benefit the whole lab, feel free to share them through OneDrive, or we can make the information available in this handbook.

### Paper lab notebooks

If you are working on sample collection or sequencing in the wet lab, you can keep records in a traditional, paper lab notebook if you prefer. Please periodically scan your notebook and save in Obsidian.

## Sapelo2 Cluster

Information on GACRC storage can be found [here](https://wiki.gacrc.uga.edu/wiki/Disk_Storage).

### Home directory
Path: /home/{user_name}
Space: 200 GB
Back up: backed up
What to store here: user specific configuration files and software installation

### Scratch directory
Path: /scratch/{user_name} 
Space: no quota
Back up: not backed up, 30 day purge
What to store here: active data analysis

### Work directory
Path: /work/tdmlab
Space: 500 GB/10,000 files
Back up: not backed up, no purge
What to store here: shared, often used resources (e.g. shared software installation, copies of shared datasets actively being analyzed)

### Project directory
Path: /project/tdmlab
Space: 1 TB
Back up: backed up, however only attached to transfer nodes
What to store here:
- in a user specific directory: data and code for unpublished projects or published projects with ongoing follow up
- shared wgs datasets with the following structure:
```
/project/tdmlab/genus_species/
	assemblies/
	annotations/
	vcfs/
	pseudogenomes/
	clustering/
	trees/
	pangenomes/
```

## GitHub
- Private repository containing collected metadata for publicly available and Mortimer lab whole genome sequencing data
- Public repositories for general assembly/analysis pipelines
- Private repositories for unpublished, ongoing projects 
- Public repositories for published projects (including preprints)
- Lab handbook and website

## OneDrive

Information on OneDrive resources provided by UGA can be found here: [OneDrive at UGA](https://ugamail.uga.edu/onedrive_business/)

### Lab shared folder
- Code and results for published projects
- Successful fellowship applications
- Sharing manuscript drafts, grant applications, abstracts, lab meeting slides (fine to use google drive here if preferred; however, only 3GB available)

### Individual folder
- Lab notebooks (shared with me)
- Ideally used as a backup for all your UGA related files

## For published projects

- Raw data should be uploaded to appropriate databases (e.g. SRA for WGS reads)
- Code and data (analysis results < 50MB) should be in a public repository on GitHub
- Code and all analysis results (not raw data if it's been uploaded to a database) should be in lab OneDrive
- Both GitHub and OneDrive directories should have a detailed README

## External Hard Drive

In addition to using OneDrive to automatically sync your files, please regularly back up your computer to an external hard drive. The lab will provide a hard drive for this purpose. 

## Data organization best practices

### Project organization

I suggest making an individual directory and git repository for each project you are working on that contains the data and scripts associated with the project, including a README.txt file describing the contents of the folder and a license text file (see https://choosealicense.com/). You should also use a workflow manager (Snakemake, Nextflow) or notebook (Jupyter, R Markdown) to document your analyses. 

Here's an example directory structure for a project using Snakemake for workflow management and conda for software installation.

```
conda_envs/
data/
   raw_data/
   analysis1/
   analysis2/
   analysis3/
figures/
LICENSE.txt
scripts/
Snakefile
README.txt
```

### File names

Please follow these rules for file names:
- Please don't use spaces, instead use underscores or hyphens
- Use descriptive names (not `figure1.txt` or `draft.docx`)
- Use dates when it makes sense. If you start your files with `YYMMDD` or `YYYY-MM-DD`, they will automatically be sorted in date order. Pick one and be consistent. Use a date rather than writing "final_FINAL-final" in your file names.

### Tables and spreadsheets

If you are creating a new dataset in a table or spreadsheet, please consider these [guidelines](https://doi.org/10.1080/00031305.2017.1375989) from Karl Broman and Kara Woo and the [Data tidying chapter](https://r4ds.hadley.nz/data-tidy.html) from R for Data Science .

# Training

## Introductory Reading
- [Microbiology primers](https://www.microbiologyresearch.org/content/microbial-primers)
- [Molecular mechanisms of antibiotic resistance revisited](https://www.nature.com/articles/s41579-022-00820-y)
- [Bacterial population genomics](https://doi.org/10.1002/9781119487845.ch36)
- [Bacterial genomic epidemiology](https://doi.org/10.1098/rstb.2021.0246)
- [*Neisseria gonorrhoeae* genomics and antimicrobial resistance](https://doi.org/10.1111/nyas.13871)

## Computational Skills

In the Mortimer Lab, we regularly use the Unix command line and slurm to interact with the cluster, git and GitHub for version control, Python for sequencing data analyses, R for statistical analyses, and Snakemake to create data analysis pipelines.

### Offered by UGA

GACRC offers extensive Unix and slurm training in order to set up a new account on the Sapelo2 cluster. 

In addition to cluster training, GACRC offers additional training on topics like Python and R. Check out the most up to date training information [here](https://wiki.gacrc.uga.edu/wiki/Training).

Training is also available through [Carpentries@UGA](https://uga-carpentries.github.io/).

### Online Articles, Tutorials, and Textbooks

#### Unix
- [Software Carpentry- The Unix Shell](https://swcarpentry.github.io/shell-novice/)

#### Git
- [Software Carpentry- Version Control with Git](https://swcarpentry.github.io/git-novice/)

#### Python
- [Software Carpentry- Programming with Python](https://swcarpentry.github.io/python-novice-inflammation/)
- [Automate the Boring Stuff with Python](https://automatetheboringstuff.com/)

#### R
- [R for Data Science](https://r4ds.had.co.nz/)
- [The Epidemiologist R Handbook](https://epirhandbook.com/en/)
- [Software Carpentry- Programming with R](http://swcarpentry.github.io/r-novice-inflammation/)

#### Snakemake
- [snakemake for doing bioinformatics series by Titus Brown](http://ivory.idyll.org/blog/tag/snakemake.html)

#### Bioinformatics and Microbial Genomics
- [Rosalind](https://rosalind.info/about/)
- [ASM Training in NGS for Infectious Disease Applications Webinars](https://asm.org/Webinars/training-ngs-infectious-diseases)
- [Guide to bacterial genome assembly by Ryan Wick](https://github.com/rrwick/Trycycler/wiki/Guide-to-bacterial-genome-assembly)
- [An applied genomic epidemiology handbook by Allison Black and Gytis Dudas](https://alliblk.github.io/genepi-book/index.html)
- [Assembling the perfect bacterial genome](https://doi.org/10.1371/journal.pcbi.1010905)
- [A scalable analytical approach from bacterial genomes to epidemiology](https://doi.org/10.1098/rstb.2021.0246)

#### Statistics
- [Statistical Rethinking by Richard McElreath](https://xcelab.net/rm/statistical-rethinking/)

### Textbooks Available in the Lab


## Biosafety

Coming soon!

## Professional Development

### Resources at UGA

- UGA's [Professional Education Portal](https://hr.uga.edu/pep/) has resources for faculty and staff training
- [Professional Photo Booth](https://career.uga.edu/professional_photo_booth) at the UGA Career Center

### Online Resources
- [PLoS Computational Biology's Ten Simple Rules articles](https://collections.plos.org/collection/ten-simple-rules/)
- [NIH OITE's Resources for Young Scientists videos](https://www.youtube.com/playlist?list=PLxnpU66KqCcjSmcKUoveWwCCfS8dlbS0R)

### Conferences

Graduate students and postdocs can expect to attend at least one conference per year funded by the lab. Trainees are expected to submit an abstract to present at conferences they plan to attend, and you should plan to apply for any travel awards provided by the conference or university if you are eligible. A final draft of the abstract must be sent to all co-authors at least one week before the submission deadline for feedback. Please have a draft to me at least two weeks in advance, so that we have time for edits before sending to co-authors.

The following are conferences that you might consider attending:
#### Microbiology
[ASM Microbe](https://asm.org/Events/ASM-Microbe/Home)
[ASM Southeastern Branch Conference](https://sebasm.org/)
[Boston Bacterial Meeting](https://bostonbacterial.org/)
#### Microbial Genomics
[Lake Arrowhead Microbial Genomics](https://www.lamg.info/)
[Applied Bioinformatics and Public Health Microbiology](https://coursesandconferences.wellcomeconnectingscience.org/)
[Microbial Population Biology Gordon Research Conference](https://www.grc.org/microbial-population-biology-conference/)
#### Antimicrobial Resistance
[Antimicrobial Resistance: Genomes, Big Data and Emerging Technologies](https://coursesandconferences.wellcomeconnectingscience.org/)
#### Sexually Transmitted Infections
[CDC STD Prevention](https://www.cdc.gov/stdconference/default.htm)
[STI & HIV World Congress](https://stihiv2025.org/)
[International Pathogenic Neisseria Conference](https://neisseria.org/ipnc/)
[NgoRS Conference](https://ngosociety.org/)
[Infectious Diseases Society for Obstetrics and Gynecology Annual Meeting](https://www.idsog.org/)
#### Evolution
[Evolution](https://www.evolutionmeetings.org/)
[Society for Molecular Biology and Evolution Meetings](https://www.smbe.org/smbe/MEETINGS.aspx)

### Funding Opportunities

#### Postdoctoral

- NIH F32
- NIH K-mechanism
- Life Science Research Foundation (LSRF) Fellowship
- HHMI Hanna H. Gray Fellowship

#### Graduate

- NSF Graduate Research Fellowship Program (GRFP)
- NIH F30/F31

#### Undergraduate

### Job Boards

#### Science job boards
[Nature Careers](https://www.nature.com/naturecareers/)
[Science Careers](https://jobs.sciencecareers.org/)
[jobRxiv](https://jobrxiv.org/)
#### Microbiology or infectious diseases job boards
[American Society for Microbiology](https://www.asmcareerconnections.org/jobs/)
[Infections Disease Dynamics Jobs](https://iddjobs.org/)
[Microbial Sciences Initiative Job Board (New England/remote)](https://msi.harvard.edu/blog)
#### Ecology and evolution job boards
[Ecoevojobs (crowdsourced google doc)](https://www.ecoevojobs.net/)

# I need help with...

## Computing on Windows

### Windows Subsystem for Linux

The easiest way to get a unix terminal on Windows is to install [Windows Subsystem for Linux](https://learn.microsoft.com/en-us/windows/wsl/install).

## Choosing a text editor or IDE

When you are working on a cluster system or in the unix shell, it can be helpful to know how to use a text editor since you may frequently need to edit files like job submission scripts without transferring back and forth from your own computer. People have [strong opinions](https://xkcd.com/378/) on which text editor is best, but you should use whichever you are most comfortable with.

### Text editors available on Sapelo2

#### Vim

[Vim](https://www.vim.org/) is the text editor that I use, so it is the one I will be able to help you with if you choose to use it.

##### Vim resources and plugins

- [vim adventures](https://vim-adventures.com/): a game that will help you learn vim keyboard shortcuts as a beginner
- If you decide to use vim, you should consider installing [neovim](https://neovim.io/), which is a refactored version of Vim that supports up-to-date syntax highlighting and plugin management.
- [Nvim-R](https://github.com/jamespeapen/Nvim-R/wiki): a plugin for editing R code

#### emacs

[Emacs](https://www.gnu.org/software/emacs/) is another popular text editor.

##### emacs resources

- [emacs manual](https://www.gnu.org/software/emacs/documentation.html)

#### nano

Nano is a basic text editor. It is probably more intuitive than vim or emacs, but it is also less powerful.

##### nano resources

- [nano manual](https://www.nano-editor.org/dist/v2.2/nano.html)

#### Open OnDemand text editor

You can also edit files in the browser using through [Open OnDemand](https://wiki.gacrc.uga.edu/wiki/OOD_Apps:_Files). This text editor has options for both vim and emacs key bindings.

### Text editors for your personal computer

You can also use vim, emacs, or nano on your personal computer if you are working in the Windows Subsystem for Linux, Linux, or Mac Terminal. Additionally, you could use a text editor with a graphical user interface, like these:

- [Sublime Text](https://www.sublimetext.com/): available for Windows, MacOS, and Linux
- [Notepad++](https://notepad-plus-plus.org/): available for Windows

### Integrated Development Environments (IDEs) and Notebooks

IDEs and notebooks allow you to interactively edit and test your code within the same software.

#### RStudio

If you regularly work in R, you might enjoy using [RStudio](https://posit.co/products/open-source/rstudio/). RStudio is available for Windows, MacOS, and Linux. It is also available on Sapelo2 through Open OnDemand. RStudio also recently added support for Python.

##### RStudio resources

- [RStudio Projects](https://support.posit.co/hc/en-us/articles/200526207-Using-RStudio-Projects): easily keep track of a project's working directory and history; easily use version control
- [R Markdown Notebooks](https://rmarkdown.rstudio.com/lesson-1.html)
- [GACRC wiki](https://wiki.gacrc.uga.edu/wiki/Rstudio-Sapelo2) for using RStudio on Sapelo2

#### Jupyter

[Jupyter](https://jupyter.org/) is another popular notebook-based IDE, which you can use on your computer or Sapelo2. You can use Jupyter for Python, R, or Julia.

##### Jupyter resources
- [GACRC wiki](https://wiki.gacrc.uga.edu/wiki/Jupyter-Sapelo2) for using Jupyter on Sapelo2


